docs/book/src/developer/guide.md

```diff
@@ -221,3 +221,12 @@ Now you can [create CAPI objects][qs]!
 To test another iteration, you'll need to follow the steps to build, push, update the manifests, and apply.
 
 [qs]: https://cluster-api.sigs.k8s.io/user/quick-start.html#usage
+
+## Videos explaining CAPI architecture and code walkthrough
+
+- CAPI components and architecture
+	- https://youtu.be/npFO5Fixqcc
```

Comment:
Consider formatting the links with the title of the video so it's easier to know which one is which when we add more to the list in the future

```suggestion
	- [Cluster API Deep Dive](https://youtu.be/npFO5Fixqcc)
```